### ISSUE TITLE
Domain variable

### ISSUE THREAD

**Issue by AramVK** [Petertjuh360/da-letsencrypt#12]
[11-Dec-2015 12:28:15 Europe/Amsterdam] PHP Notice:  Undefined variable: domain in /usr/local/directadmin/plugins/da-letsencrypt/user/index.html on line 10
[11-Dec-2015 12:28:15 Europe/Amsterdam] PHP Notice:  Undefined variable: domain in /usr/local/directadmin/plugins/da-letsencrypt/user/index.html on line 14

```
    <input type="hidden" name="domain" value="<?= $domain; ?>"/>
```

Also I'm not sure, but is the domain going through these kind of checks?
- valid domain, no weird characters
- do you own the domain, not some other user

**Comment by zenire** [comment]
Could you edit /user/index.html and paste the following code on the second line?

`<? parse_str(getenv('QUERY_STRING')); ?>`

Try again, please.

**Comment by Wouter0100** [comment]
I saw this issue in the morning, yeah. Had no time to fix it tho.

`$domain` must be changed to `$_POST['domain']`. I've switched everything to the regular global variables (from _POST and _GET). This because of security, an user is able to define all variables he/she wants, if we parse it to local variables, instead of an array.

If you want to push a PR, you're more then welcome :+1:

**Comment by zenire** [comment]
@Wouter0100 Does DirectAdmin submit a POST with 'domain' to index.html when you visit the plugin?

**Comment by Wouter0100** [comment]
@Petertjuh360 No, [but we make it work](https://github.com/Petertjuh360/da-letsencrypt/blob/master/includes/bootstrap.php#L10-L11).

**Comment by zenire** [comment]
@Wouter0100 Thank you! Smart

**Comment by AramVK** [comment]
Maybe for validation the file `/usr/local/directadmin/data/users/<user>/domains.list` could be consulted? I don't know if DA offers any information about it in a plugin environment

**Comment by Wouter0100** [comment]
That's unfortunately not possible, because we don't have read access to that file from the `request.html`.

**Comment by AramVK** [comment]
Isn't it all diradmin:diradmin?

**Comment by Wouter0100** [comment]
Yeah, but the file is executed as the user.

**Comment by zenire** [comment]
We could run it as the directadmin user but that is a security issue.

**Comment by zenire** [comment]
https://www.directadmin.com/features.php?id=1594

**Comment by Wouter0100** [comment]
Can't we use the API for this? Internal it's very fast, which should not really slow the plugin down. We're then able to not run as another user.

**Comment by AramVK** [comment]
Just found this one http://www.directadmin.com/features.php?id=1410 
It's in `$_SERVER['SESSION_SELECTED_DOMAIN']`

It might not always be set it says so the api way could be solid, but maybe this could be enough, it only seems to be updating if it's a valid domain you're accessing.

**Comment by Wouter0100** [comment]
Hmm, that feature seems awesome for this. Then we shouldn't even need to pass it in the URL, don't we?

**Comment by zenire** [comment]
That feature should be the solution, I think.

**Comment by AramVK** [comment]
Yep seems like it!

**Comment by Wouter0100** [comment]
Implemented in ac8c41fac09614655fd94b374c451093bc91f9e1
